packages/message-box/src/index.vue

```diff
@@ -0,0 +1,369 @@
+      popup.state.visible = false
+      popup.updateClosingFlag(true)
+
+      vm.ctx.onClose && vm.ctx.onClose()
+      dialog.closeDialog() // 解绑
+      if (props.lockScroll) {
+        setTimeout(popup.restoreBodyStyle, 200)
+      }
+      popup.state.opened = false
+      popup.doAfterClose()
+      setTimeout(() => {
+        if (state.action) state.callback(state.action, vm.ctx)
+      })
+    }
+
+    const getFirstFocus = () => {
+      const btn = vm.vnode.el.querySelector('.el-message-box__btns .el-button')
+      const title = vm.vnode.el.querySelector('.el-message-box__btns .el-message-box__title')
+      return btn || title
+    }
+
+    const handleWrapperClick = () => {
+      if (props.closeOnClickModal) {
+        handleAction(state.distinguishCancelAndClose ? 'close' : 'cancel')
+      }
+    }
+
+    const handleInputEnter = () => {
+      if (state.inputType !== 'textarea') {
+        return handleAction('confirm')
+      }
+    }
+
+    const handleAction = action => {
+      if (state.type$ === 'prompt' && action === 'confirm' && !validate()) {
+        return
+      }
+      state.action = action
+      if (typeof vm.proxy.beforeClose === 'function') {
+        vm.proxy.close = getSafeClose()
+        vm.proxy.beforeClose(action, vm.proxy, vm.proxy.close)
+      } else {
+        doClose()
+      }
+    }
+
+    const validate = () => {
+      if (state.type$ === 'prompt') {
+        const inputPattern = state.inputPattern
+        if (inputPattern && !inputPattern.test(state.inputValue || '')) {
+          state.editorErrorMessage = state.inputErrorMessage || t('el.messagebox.error')
+          addClass(getInputElement(), 'invalid')
+          return false
+        }
+        const inputValidator = state.inputValidator
+        if (typeof inputValidator === 'function') {
+          const validateResult = inputValidator(state.inputValue)
+          if (validateResult === false) {
+            state.editorErrorMessage = state.inputErrorMessage || t('el.messagebox.error')
+            addClass(getInputElement(), 'invalid')
+            return false
+          }
+          if (typeof validateResult === 'string') {
+            state.editorErrorMessage = validateResult
+            addClass(getInputElement(), 'invalid')
+            return false
+          }
+        }
+      }
+      state.editorErrorMessage = ''
+      removeClass(getInputElement(), 'invalid')
+      return true
+    }
+
+    const getInputElement = () => {
+      const inputRefs = vm.refs.input.$refs
+      return inputRefs.input || inputRefs.textarea
+    }
+
+    const handleClose = () => {
+      handleAction('close')
+    }
+
+    return {
+      ...toRefs(state),
+      ...toRefs(popup.state),
+      icon,
+      confirmButtonClasses,
+      cancelButtonClasses,
+      handleWrapperClick,
+      handleInputEnter,
+      handleAction,
+      handleClose,
+      t,
+      doClose,
+    }
+  },
+})
+</script>
+<style scoped>
```

Comment:
The component's style shouldnot have `scoped style`

Reply:
done